rfcs/taskforce/0002-network-utility.md

```diff
@@ -0,0 +1,244 @@
+
+## Command-line Framework
+
+After investigating the current state of commandline infrastructure I've come up with the following three options listed in priority order:
+
+### Cmdliner
+
+This is an OCaml project that a variety of members of our engineering team have experience with. This framework supports a variety of features out of the box that we'd like to have such as:
+
+1. Robust Argument Parsing
+2. Subcommands
+3. Documentation support
+4. Strongly typed with Reason support
+
+Given all these benefits and existing Bucklescript integration, this seems like the best solution to get going quickly and allowing other members of the team to contribute without having to learn a new tool.
+
+**Links**
+
+* Original project - [github](https://github.com/dbuenzli/cmdliner), [website](https://erratique.ch/software/cmdliner)
+* Bucklescript bindings - [github](https://github.com/ELLIOTTCABLE/bs-cmdliner)
+
+### Ink
+
+React based framework for building command-line tools. While this project seems to have a ton of functionality, it seems like it might be a bit overkill for this project. There's no Reason bindings, however they could be ported over reasonably easily from the TypeScript definitions. Overall this seems like a strong contender however is more work than cmdliner and would require everyone to learn a new tool.
+
+**Links**
+
+* Ink - [github](https://github.com/vadimdemedes/ink)
+
+### Custom Implementation
+
+This was also a strong option since the amount of functionality we wish to support is well scoped and could be implemented by hand without too much work. The benefits of this approach are dropping a dependency and allowing us to only implement the features that we plan to use. 
+
+Downside of this approach is needing to maintain functionality that is reasonably well supported and maintained by the previously mentioned projects.
+
+## Use Cases
+
+There are three target use cases for this tool. The main use case is sharing the required configuration and keyfiles for deploying testnets. Secondarily the programatic API would enable integrations into other flows (such as interation testing). Finally the tool should be useful for monitoring the health of a live network.
+
+### Deploying Testnets
+
+When we need to deploy a testnet, we first need to collect all the keysets needed for inclusion in the genesis ledger of the new network.
+
+1. Generate new keys for "whale" / "fish" keysets
+2. Re-use keys for "services" keyset
+3. Gather keys from community with discord info
+
+This is accomplished with the `coda keyset` commands.
+
+Once we have the keys we can generate the genesis ledger, for each keyset the keys will have the following state:
+
+* Initial account balance
+* Initial delegate status
+
+This happens with the `coda genesis` commands.
+
+Finally the configuration can be deployed to the kubernetes environment for use in a deployment using the `coda publish` command.
+
+### Programmatic API
```

Comment:
can you include one little section going something like: "For O(1)'s management of testnets: we can use this API to automate most of the Google-form genesis signup -> genesis ledger creation, and for integration testing dynamic reconfiguration of keysets as desired by specific tests"